packages/astro/src/content/template/types.d.ts

```diff
@@ -48,6 +48,10 @@ declare module 'astro:content' {
 		collection: C,
 		filter?: (entry: CollectionEntry<C>) => entry is E
 	): Promise<E[]>;
+	export function getCollection<C extends keyof typeof entryMap>(
+		collection: C,
+		filter?: (data: CollectionEntry<C>) => unknown
```

Comment:
```suggestion
		filter?: (data: CollectionEntry<C>) => boolean
```
Not sure if this breaks type inferencing? Seems odd to require `unknown`.

Reply:
I don't think it _requires_ `unknown`, it allows you to return any value rather than only `boolean`, just like a native `filter`.

Reply:
got it! In that case, approved ❤️

Reply:
Right. `boolean` is a subtype of `unknown`, and native `filter` [allows](https://github.com/microsoft/TypeScript/blob/v4.9.4/lib/lib.es5.d.ts#L1442-L1453) `unknown`.